### PR TITLE
Debs support multiple foreman exports

### PR DESCRIPTION
**Pull request by nahiluhmot**
@bfulton @adamjt @tlunter

### CODE REVIEW

**Comment by adamjt** [comment]
If I create two foreman blocks with the same name, this will stomp on all but the last block, right?

**Comment by adamjt** [comment]
Besides that, I'm very excited to have this.

**Comment by nahiluhmot** [comment]
@adamjt That's correct, it will stomp on it. That being said, that's the case for any two DSL objects that share a name and class.

**Comment by adamjt** [comment]
Makes sense to me!

**Comment by bfulton** [comment]
Great stuff! Why would you have to specify different build dirs for each foreman block?

**Comment by nahiluhmot** [comment]
@bfulton Because the `Foreman` class [removes its build dir](https://github.com/swipely/dockly/blob/a8e2271068b933ba64466799f22652f3dc3bed84/lib/dockly/foreman.rb#L22) at the beginning of builds. I wanted to change `build_dir` to `build_dir_prefix` and use the unique `name` attribute to make a unique directory, but that would be a backwards-incompatible change.

File: lib/dockly/deb.rb [hunk 1]
```diff
@@ -127,14 +127,14 @@ def convert_package
...
-    foreman.create!
-    package.attributes[:prefix] = foreman.init_dir
-    Dir.chdir(foreman.build_dir) do
-      package.input('.')
+    return if (foreman || []).empty?
+    foreman.each do |fore|
+      info "adding foreman export '#{fore.name}'"
+      fore.create!
+      package.attributes[:prefix] = fore.init_dir
+      Dir.chdir(fore.build_dir) { package.input('.') }
```
Comment (tlunter) [inline lib/dockly/deb.rb]: Does this just stage a copy when we convert from a Dir package to a Deb/RPM package?  It doesn't copy immediately?

Comment (tlunter) [inline lib/dockly/deb.rb]: If it copies immediately, do we still need a separate build dir?

Comment (nahiluhmot) [inline lib/dockly/deb.rb]: @tlunter Good question, will validate by setting the same build dir in the tests. If so, will remove the warning in the README

**Comment by nahiluhmot** [comment]
@tlunter Updated -- looks like they need not be separate.

**Comment by bfulton** [comment]
+1

**Comment by tlunter** [comment]
Nice. :+1:

**Pull request merged**